app/services/match_orders.rb

```diff
@@ -0,0 +1,46 @@
+class MatchOrders
```

Comment:
Is MatchOrders the best name?

Is there some stock exchange terminology that better describes what is happenening? Match conveys finding pairs but not actually doing anything with it.